velox/functions/sparksql/String.h

```diff
@@ -1130,4 +1130,59 @@ struct ReplaceFunction {
   }
 };
 
+template <typename T>
+struct FindInSetFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool callNullFree(
+      out_type<int32_t>& result,
+      const null_free_arg_type<Varchar>& str,
+      const null_free_arg_type<Varchar>& strArray) {
+    return doCall(result, str, strArray);
+  }
+
+  FOLLY_ALWAYS_INLINE bool
+  doCall(out_type<int32_t>& result, StringView match, StringView strArray) {
+    int32_t index = 1;
+    int32_t lastComma = -1;
+
+    auto _match = match.str();
+    auto _strArray = strArray.str();
+
+    auto found = _match.find(',');
+    if (found != std::string::npos) {
+      result = 0;
+      return true;
+    }
+
+    auto _strArrayData = _strArray.c_str();
+    size_t _strArraySize = _strArray.size();
+    size_t _matchSize = _match.size();
+
+    for (int i = 0; i < _strArraySize; i++) {
+      if (_strArrayData[i] == ',') {
+        auto substr = _strArray.substr(lastComma + 1, i - (lastComma + 1));
```

Comment:
`substr` will generate a lot of small strings, the number depends on which depends on the number of `,` , maybe https://[cplusplus.com/reference/string/string/compare/](https://cplusplus.com/reference/string/string/compare/) can help you.

Reply:
Fixed, thanks